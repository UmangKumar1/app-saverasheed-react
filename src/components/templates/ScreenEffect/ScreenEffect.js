import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { CSSTransition } from 'react-transition-group'
import styled, { keyframes, cx, css } from 'react-emotion'

const { any, bool, string } = PropTypes

const easeOutQuint = 'cubic-bezier(0.230, 1.000, 0.320, 1.000)'
const easeInQuint = 'cubic-bezier(0.755, 0.050, 0.855, 0.060)'
const screenBackground = '#121010'

export const turnOn = keyframes`
0% {
  transform: scale(1,0.8) translate3d(0,0,0);
  filter: brightness(30);
  opacity: 1;
}

28% {
  transform: scale(1,0.8) translate3d(0,100%,0);
}

28.8% {
  transform: scale(1,0.8) translate3d(0,-100%,0);
  opacity: 1;
} 

72% {
  transform: scale(1.3,0.6) translate3d(0,100%,0);
  filter: brightness(30);
  opacity: 0;
}

88% {
  transform: scale(1,1) translate3d(0,0,0);
  filter: contrast(0) brightness(0);
  opacity: 1;
}

100% {
  transform: scale(1,1) translate3d(0,0,0);
  filter: contrast(1) brightness(1.2) saturate(1.3);
  opacity: 1;
}
`

const turnOff = keyframes`
0% {
  transform: scale(1,1.3) translate3d(0,0,0);
  filter: brightness(1);
  opacity: 1;
}
60% {
  transform: scale(1.3,0.001) translate3d(0,0,0);
  filter: brightness(10);
}
100% {
  animation-timing-function: ${easeInQuint};
  transform: scale(0.000,0.0001) translate3d(0,0,0);
  filter: brightness(50);
}
`

const ScreenEffectContainer = styled('div')({
  background: screenBackground,
  width: '100%',
  height: '100%',
  position: 'relative',
  overflow: 'auto',
})

const screenCx = css({
  width: '100%',
  height: '100%',
  border: 'none',
  '&-enter-active': {
    animation: `${turnOn} 300ms linear`,
    animationFillMode: 'forwards',
  },
  '&-exit-active': {
    animation: `${turnOff} 550ms ${easeOutQuint}`,
    animationFillMode: 'forwards',
  },
})

const ScreenEffect = ({ children, on, showAnim, className = '', ...props }) => (
  <ScreenEffectContainer
    className={cx('screen-effect-container', className)}
    {...props}
  >
    <CSSTransition
      in={on}
      timeout={{
        enter: 300,
        exit: 550,
      }}
      unmountOnExit
      classNames={screenCx}
    >
      <Fragment>{children}</Fragment>
    </CSSTransition>
  </ScreenEffectContainer>
)

ScreenEffect.propTypes = {
  children: any.isRequired,
  on: bool.isRequired,
  className: string,
}

export default ScreenEffect
