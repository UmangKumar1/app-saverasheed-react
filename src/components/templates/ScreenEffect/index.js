export { default, default as ScreenEffect } from './ScreenEffect'
