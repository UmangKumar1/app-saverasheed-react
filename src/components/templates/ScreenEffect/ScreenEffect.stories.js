import React, { Component } from 'react'
import { storiesOf } from '@storybook/react'
import styled from 'react-emotion'
import ScreenEffect from '.'

const ScreenEffectRoot = styled('div')({
  height: '100%',
  width: '100vw',
  background: '#111',
  color: 'white',
  padding: 20,
})

const Switch = styled('input')({
  display: 'none',
})

const Label = styled('label')(
  {
    display: 'inline-block',
    cursor: 'pointer',
    background: '#fff',
    color: '#111',
    padding: 10,
    paddingLeft: 15,
    paddingRight: 15,
    borderRadius: 5,
    marginTop: 10,
    boxShadow: '0 2px #666',
    '&::before': {
      content: '" "',
      display: 'inline-block',
      borderRadius: '100%',
      width: 10,
      height: 10,
      background: '#003321',
      marginRight: 10,
      border: '1px solid #111',
    },
  },
  ({ active }) =>
    active && {
      '&::before': {
        background: '#2f5',
      },
    },
)

class ScreenEffectStory extends Component {
  state = { on: this.props.on }

  render() {
    const { on } = this.state
    return (
      <ScreenEffectRoot>
        <Switch
          type="checkbox"
          id="switch"
          checked={on}
          onChange={() => this.setState(({ on }) => ({ on: !on }))}
        />
        <Label htmlFor="switch" active={on}>
          Turn {on ? 'off' : 'on'}
        </Label>
        <ScreenEffect on={on}>foo bar bin baz</ScreenEffect>
      </ScreenEffectRoot>
    )
  }
}

class ScreenEffectChildrenStory extends Component {
  state = { on: true }

  render() {
    const { on } = this.state
    return (
      <ScreenEffectRoot>
        <Switch
          type="checkbox"
          id="switch"
          checked={on}
          onChange={() => this.setState(({ on }) => ({ on: !on }))}
        />
        <Label htmlFor="switch" active={on}>
          Turn {on ? 'off' : 'on'}
        </Label>
        <ScreenEffect on={on}>
          <img src="https://placekitten.com/g/600/600" />
        </ScreenEffect>
      </ScreenEffectRoot>
    )
  }
}

storiesOf('ScreenEffect', module)
  .add('on', () => <ScreenEffectStory on />)
  .add('off', () => <ScreenEffectStory />)
  .add('with complex children', () => <ScreenEffectChildrenStory />)
