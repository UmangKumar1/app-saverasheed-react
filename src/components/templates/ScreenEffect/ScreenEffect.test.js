import React from 'react'
import { shallow } from 'enzyme'
import ScreenEffect from '.'

const wrap = (props = {}) => shallow(<ScreenEffect {...props} />)

describe('ScreenEffect', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ children: 'ScreenEffect', on: false })
    expect(wrapper).toMatchSnapshot()
  })
})
