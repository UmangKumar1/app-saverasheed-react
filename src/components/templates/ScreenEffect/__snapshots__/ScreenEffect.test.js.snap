// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenEffect renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScreenEffect
    on={false}
  >
    ScreenEffect
  </ScreenEffect>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <CSSTransition
        classNames="css-yi9g2j"
        in={false}
        timeout={
          Object {
            "enter": 300,
            "exit": 550,
          }
        }
        unmountOnExit={true}
      >
        <React.Fragment>
          ScreenEffect
        </React.Fragment>
      </CSSTransition>,
      "className": "screen-effect-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <React.Fragment>
          ScreenEffect
        </React.Fragment>,
        "classNames": "css-yi9g2j",
        "in": false,
        "timeout": Object {
          "enter": 300,
          "exit": 550,
        },
        "unmountOnExit": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "ScreenEffect",
        },
        "ref": null,
        "rendered": "ScreenEffect",
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <CSSTransition
          classNames="css-yi9g2j"
          in={false}
          timeout={
            Object {
              "enter": 300,
              "exit": 550,
            }
          }
          unmountOnExit={true}
        >
          <React.Fragment>
            ScreenEffect
          </React.Fragment>
        </CSSTransition>,
        "className": "screen-effect-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <React.Fragment>
            ScreenEffect
          </React.Fragment>,
          "classNames": "css-yi9g2j",
          "in": false,
          "timeout": Object {
            "enter": 300,
            "exit": 550,
          },
          "unmountOnExit": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "ScreenEffect",
          },
          "ref": null,
          "rendered": "ScreenEffect",
          "type": Symbol(react.fragment),
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
